### PR TITLE
Upgrade gRPC version

### LINKED ISSUE #3467: Cleanup: Synchronise gRPC versions across the repository
Thanks to dependabot, I noticed we have both gRPC `v1.57.0` and `v1.53.0` in a variety of places.

* https://github.com/search?q=repo%3Agoogleforgames%2Fagones+%221.57.0%22&ype=code
* https://github.com/search?q=repo%3Agoogleforgames%2Fagones+%221.53.0%22&type=code

We should move it all to 1.57.1 (as depdendabot would like us to do).

Thankfully we have a checklist for this for Kubernetes updates, so I'll steal that:

- [ ] Update the SDK [base image grpc version](https://github.com/googleforgames/agones/blob/main/build/includes/sdk.mk#L30) and rebuild the image. Note that this can take a while and in the past we have had to manually push it to gcr because cloud build doesn't like how long it takes.
- [ ] Regenerate allocated API endpoints: [make gen-allocation-grpc](https://github.com/googleforgames/agones/blob/main/build/includes/allocation.mk#L55)
- [ ] Regenerate all client sdks: [make gen-all-sdk-grpc](https://github.com/googleforgames/agones/blob/main/build/README.md#make-gen-all-sdk-grpc)
- [ ] Update the version number in C++ Cmake scripts [here](https://github.com/googleforgames/agones/blob/main/sdks/cpp/CMakeLists.txt#L100) and [here](https://github.com/googleforgames/agones/blob/main/sdks/cpp/cmake/prerequisites.cmake#L34)

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3467 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by Kalaiselvi84** [comment]
@markmandel, Could you please guide me how to fix this error?
```
CMake Error at CMakeLists.txt:32 (_message):
  CMake Error: The source directory
  "/go/src/agones.dev/agones/sdks/cpp/.build/gRPC/src/third_party/protobuf/cmake"
  does not appear to contain CMakeLists.txt.

  Specify --help for usage, or press the help button on the CMake GUI.

Call Stack (most recent call first):
  cmake/prerequisites.cmake:93 (message)
  cmake/prerequisites.cmake:114 (execute_and_check)
  cmake/prerequisites.cmake:174 (invoke_cmake_build)
  CMakeLists.txt:73 (include)


-- Configuring incomplete, errors occurred!
See also "/go/src/agones.dev/agones/sdks/cpp/.build/CMakeFiles/CMakeOutput.log".
See also "/go/src/agones.dev/agones/sdks/cpp/.build/CMakeFiles/CMakeError.log".
make: *** [Makefile:29: build] Error 1
make[1]: *** [includes/sdk.mk:88: run-sdk-command] Error 2
make: *** [includes/sdk.mk:72: run-sdk-command-cpp] Error 2
```

**Comment by markmandel** [comment]
Will pull this down and take a look 👍🏻

**Comment by markmandel** [comment]
So just digging into this - looks like you may have run `make gen-sdk-grpc` (which defaults to `go`) not `make gen-all-sdk-grpc` since only the Go code got updated (also I get extra updates when I run it locally). Easy thing to do as the commands are next to each other -- we should probably turn off the default for `gen-sdk-grpc`.

if you run `make gen-all-sdk-grpc` you should see changes that look more like this:

```shell
❯ git shell
On branch pr/grpc-update
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   pkg/sdk/alpha/alpha.pb.go
        modified:   pkg/sdk/alpha/alpha_grpc.pb.go
        modified:   pkg/sdk/beta/beta.pb.go
        modified:   pkg/sdk/beta/beta_grpc.pb.go
        modified:   pkg/sdk/sdk.pb.go
        modified:   pkg/sdk/sdk_grpc.pb.go
        modified:   sdks/cpp/include/agones/sdk.pb.h
        modified:   sdks/cpp/include/google/api/annotations.pb.h
        modified:   sdks/cpp/include/google/api/http.pb.h
        modified:   sdks/cpp/include/protoc-gen-openapiv2/options/annotations.pb.h
        modified:   sdks/cpp/include/protoc-gen-openapiv2/options/openapiv2.pb.h
        modified:   sdks/cpp/src/agones/sdk.pb.cc
        modified:   sdks/cpp/src/google/annotations.pb.cc
        modified:   sdks/cpp/src/google/http.pb.cc
        modified:   sdks/cpp/src/protoc-gen-openapiv2/annotations.pb.cc
        modified:   sdks/cpp/src/protoc-gen-openapiv2/openapiv2.pb.cc
```

Not 100% sure it solves the problem yet, but just working step by step.

**Comment by Kalaiselvi84** [comment]
Log for `make gen-all-sdk-grpc` - https://gist.github.com/Kalaiselvi84/68c31b262ca04b5b57275f049e438232

You're right; I may have executed `make gen-sdk-grpc`. 6 files were modified locally for the command `make gen-all-sdk-grpc`, but it seems more files were altered in yours.🤔

**Comment by markmandel** [comment]
I got the same error anyway with all the files updated, but I _think_ i have a fix.... lemme try it though.

**Comment by markmandel** [comment]
Fairly sure I got it... I'll my usual file a PR against your branch thing.

**Comment by markmandel** [comment]
I cleaned my build and am running it again - so probably have confirmation tomorrow.

**Comment by markmandel** [comment]
https://github.com/Kalaiselvi84/agones/pull/326 submitted to your branch!

The CPP tests passed for me 👍🏻

**Comment by Kalaiselvi84** [comment]
> I got the same error anyway with all the files updated, but I _think_ i have a fix.... lemme try it though.

I am confused. What's the difference between my local setup and yours? Why aren't all the files being modified on my end as they are on yours?

**Comment by markmandel** [comment]
> I am confused. What's the difference between my local setup and yours? Why aren't all the files being modified on my end as they are on yours?

I figure it's because I had rebuilt all my Docker images, so it pulled down the latest protobuf (which is why the version updated). Also because I ran the `make gen-allocation-grpc` command.

**Comment by Kalaiselvi84** [comment]
> I figure it's because I had rebuilt all my Docker images, so it pulled down the latest protobuf (which is why the version updated). Also because I ran the `make gen-allocation-grpc` command.

When you mentioned rebuilding all the Docker images, did you mean only for the example images or did you rebuild everything? Also I guess you ran `make build` instead `make build-sdks`

**Comment by markmandel** [comment]
> When you mentioned rebuilding all the Docker images, did you mean only for the example images or did you rebuild everything?

I occasionally delete all docker images from my machines just to keep them clean, and so I was rebuilding everything from scratch.

Either way - merge my PR to your PR when ready, and we'll see if it passes all of CI 👍🏻

**Comment by Kalaiselvi84** [comment]
> rebuilding everything from scratch.

Do we have any command for this?

> Either way - merge my PR to your PR when ready, and we'll see if it passes all of CI 👍🏻

This is kind of no fair to always use your updates, would it be possible for you to create a separate PR for it? What do you think? 

else share me your logs, I will try to replicate yours

**Comment by markmandel** [comment]
> Do we have any command for this?

I just do the command above occasionally when I do `docker images` and there are a lot of them there 😄 

> This is kind of no fair to always use your updates, would it be possible for you to create a separate PR for it? What do you think?

I don't mind either way - whatever works!

**Comment by markmandel** [comment]
Also just realised I was typing on my phone above and wrote `make gen-allocation-grpc` when I meant `make gen-allocation-grpc` 🤦🏻

**Comment by Kalaiselvi84** [comment]
> Also just realised I was typing on my phone above and wrote `make gen-allocation-grpc` when I meant `make gen-allocation-grpc`

This was also typing from your phone? you typed the same command here😂

**Comment by markmandel** [comment]
> > Also just realised I was typing on my phone above and wrote `make gen-allocation-grpc` when I meant `make gen-allocation-grpc`
> 
> This was also typing from your phone? you typed the same command here😂

OMG I did it again.  `make gen-all-sdk-grpc` I can do it!

**Comment by markmandel** [comment]
As part of this PR can we update `make gen-all-sdk-grpc` to show what the differences are?

**Comment by Kalaiselvi84** [comment]
> As part of this PR can we update `make test-gen-all-sdk-grpc` to show what the differences are?

`make test-gen-all-sdk-grpc` is running, will update the status shortly

**Comment by markmandel** [comment]
> > As part of this PR can we update `make test-gen-all-sdk-grpc` to show what the differences are?
> 
> `make test-gen-all-sdk-grpc` is running, will update the status shortly

Don't we need that update in this PR? We don't know what is being generated by Cloud Build and how it's different to what we have right now.

**Comment by Kalaiselvi84** [comment]
200% credit goes to @markmandel 😀

File: build/includes/sdk.mk [hunk 1]
```diff
@@ -20,14 +20,15 @@
 #   \__, |_| \_\_|    \____|   |_|\___/ \___/|_|_|_| |_|\__, |
 #   |___/                                               |___/
 
-build_sdk_base_version = $(call sha,$(build_path)/build-sdk-images/tool/base/Dockerfile)
+grpc_release_tag = v1.57.0
+
+build_sdk_base_version = $(call sha,$(build_path)/build-sdk-images/tool/base/Dockerfile)_$(grpc_release_tag)
```
Comment (markmandel) [inline build/includes/sdk.mk:25]: This now forces a rebuild when the grpc version is updated. Before this wasn't happening.

**Review by zmerlynn — APPROVED** [review]

**Review by gongmax — APPROVED** [review]

**Pull request merged**